pay_ios/lib/src/widgets/apple_pay_button.dart

```diff
@@ -133,10 +133,8 @@ class RawApplePayButton extends StatelessWidget {
 }
 
 /// A widget to draw the Apple Pay button through a [PlatforView].
-class _UiKitApplePayButton extends StatelessWidget {
+class _UiKitApplePayButton extends StatefulWidget {
```

Comment:
I believe that moving to a `StatefulWidget` (and adding a `State`) is all we need to have the reference to `onPressed` mutate when the object is updated, and hence we can keep the instantiation of the `MethodChannel` and setting the handler inside `onPlatformViewCreated`.
Can you confirm that this is the case?